lib/Twig/Template.php

```diff
@@ -607,15 +618,33 @@ protected function getAttribute($object, $item, array $arguments = array(), $typ
             $this->env->getExtension('Twig_Extension_Sandbox')->checkMethodAllowed($object, $method);
         }
 
-        // Some objects throw exceptions when they have __call, and the method we try
-        // to call is not supported. If ignoreStrictCheck is true, we should return null.
-        try {
-            $ret = call_user_func_array(array($object, $method), $arguments);
-        } catch (BadMethodCallException $e) {
-            if ($call && ($ignoreStrictCheck || !$this->env->isStrictVariables())) {
-                return;
+        if ($call) {
+            // Some objects throw exceptions when they have __call, and the method we try
+            // to call is not supported. If ignoreStrictCheck is true, we should return null.
+            try {
+                $ret = $object->__call($method, $arguments);
+            } catch (BadMethodCallException $e) {
+                if ($ignoreStrictCheck || !$this->env->isStrictVariables()) {
+                    return;
+                }
+                throw $e;
             }
-            throw $e;
+        } elseif ($a = $arguments) {
```

Comment:
yes, yet not calling count when not required is another optim *vs* handling in the switch